Python/flowgraph.c

```diff
@@ -1384,6 +1393,56 @@ fold_tuple_on_constants(PyObject *const_cache,
     return SUCCESS;
 }
 
+#define MIN_CONST_SEQUENCE_SIZE 3
+/* Replace LOAD_CONST c1, LOAD_CONST c2 ... LOAD_CONST cN, BUILD_LIST N
+   with BUILD_LIST 0, LOAD_CONST (c1, c2, ... cN), LIST_EXTEND 1,
+   or BUILD_SET & SET_UPDATE respectively.
+*/
+static int
+optimize_build_list_or_set_with_constants(PyObject *const_cache, cfg_instr* inst, int n, PyObject *consts)
```

Comment:
I'd call this `optimize_if_const_list_or_set` to indicate that it may or may not apply the optimization.